### PR TITLE
feat: use order displayState instead of state

### PR DESCRIPTION
**Pull request by mdole**
The type of this PR is: **Feat**

### Description

Companion PR to https://github.com/artsy/eigen/pull/8275. Pretty sure this should cover all of the places we're using order `state` to determine what to display/how to route users where having state be `in_review` might cause things to break.

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Order/redirects.tsx [hunk 1]
```diff
@@ -258,6 +260,7 @@ graphql`
     internalID
     mode
     state
```
Comment (iskounen) [inline src/Apps/Order/redirects.tsx:262]: **question:** Can we remove this field, or does Force still need to know about it?

Comment (iskounen) [reply src/Apps/Order/redirects.tsx:262]: I forgot to check the tests before writing this comment - it makes sense now that you're using both `state` and `displayState`.

**Review by iskounen — DISMISSED** [review]

**Comment by mdole** [comment]
@iskounen might as well! I had originally left that + one other instance where it _kinda_ made sense to use `state`, but `displayState` also works. Just updated.

I find it a little confusing now to know when we should use `state` vs. `displayState` 🤔 like I guess Force should mostly rely on `displayState`, especially any time we might need to handle `in_review` orders. But `displayState` also folds in some shipping logic...

For now, I don't think we're at too much risk of shooting ourselves in the feet, but something to keep an eye on for sure.

**Comment by mdole** [comment]
We discussed on [Slack](https://artsy.slack.com/archives/C047EHC5CSK/p1678448408172429?thread_ts=1678446811.047119&cid=C047EHC5CSK) and via Zoom and decided to keep this simple for now and only change the places we knew we needed to make a change.

**Pull request merged**